app/src/main/java/ar/com/wolox/android/features/auth/login/LoginPresenter.kt

```diff
@@ -0,0 +1,68 @@
+package ar.com.wolox.android.features.auth.login
+
+import ar.com.wolox.android.example.network.builder.networkRequest
+import ar.com.wolox.android.example.network.repository.UserRepository
+import ar.com.wolox.android.utils.UserSession
+import ar.com.wolox.android.extfunctions.isValidEmail
+import ar.com.wolox.android.models.LoginBody
+import ar.com.wolox.android.models.User
+import ar.com.wolox.android.utils.Errors
+import ar.com.wolox.wolmo.core.presenter.CoroutineBasePresenter
+import kotlinx.coroutines.launch
+import javax.inject.Inject
+
+class LoginPresenter @Inject constructor(
+    private val userSession: UserSession,
+    private val userRepository: UserRepository
+) : CoroutineBasePresenter<LoginView>() {
+
+    fun onLogInButtonClick(email: String, password: String) {
+
+        val totalErrors: MutableList<Errors> = ArrayList()
+
+        if (email.isEmpty()) {
```

Comment:
can you improve this check? i don't know, but there is something here that i think that can improve